### PR TITLE
identify all statements that can be EXPLAINed

### LINKED ISSUE #1963: Identify all statements that can be explained
The `EXPLAIN` docs current state that only `INSERT`, `SELECT`, `DELETE`, and `UPDATE` are explainable. That's an incomplete list.

For 1.0: https://www.cockroachlabs.com/docs/v1.0/sql-grammar.html#explainable_stmt
For 1.1: https://www.cockroachlabs.com/docs/v1.1/sql-grammar.html#explainable_stmt

### PR DESCRIPTION
**Pull request by lhirata**
Closes #1963.

### CODE REVIEW

File: v1.0/explain.md [hunk 1]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
 <div id="toc"></div>
 
+## Explainable statements
+
+You can use `EXPLAIN` on the following statements:
+
+- [`ALTER TABLE`](alter-table.html)
+- [`CREATE`](sql-grammar.html#create_stmt)
+- [`DELETE`](delete.html)
+- [`HELP`](sql-grammar.html#help_stmt)
```
Comment (knz) [inline v1.0/explain.md]: Remove HELP, it was not meant to be documented.

File: v1.0/explain.md [hunk 2]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`INSERT`](insert.html)
+- [`SELECT`](select.html)
+- [`SHOW`](sql-grammar.html#show_stmt)
+- [`SPLIT`](sql-grammar.html#split_stmt)
```
Comment (knz) [inline v1.0/explain.md]: There is no "SPLIT" statement, it's a special form of ALTER.

File: v1.0/explain.md [hunk 3]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable statements
+
+You can use `EXPLAIN` on the following statements:
+
+- [`ALTER TABLE`](alter-table.html)
```
Comment (knz) [inline v1.0/explain.md]: I think all the forms of ALTER can be explained, not just ALTER TABLE. Or perhaps that was not true in 1.0?

File: v1.0/explain.md [hunk 4]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`CREATE`](sql-grammar.html#create_stmt)
+- [`DELETE`](delete.html)
+- [`HELP`](sql-grammar.html#help_stmt)
+- [`INSERT`](insert.html)
+- [`SELECT`](select.html)
+- [`SHOW`](sql-grammar.html#show_stmt)
+- [`SPLIT`](sql-grammar.html#split_stmt)
+- [`UPDATE`](update.html)
```
Comment (knz) [inline v1.0/explain.md]: You can add EXPLAIN to this list too (EXPLAIN is explainable)

File: v2.0/explain.md [hunk 5]
```diff
@@ -4,10 +4,32 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`PAUSE JOB`](pause-job.html)
+- [`RESET`](sql-grammar.html#reset_stmt)
+- [`RESTORE`](restore.html)
+- [`RESUME JOB`](resume-job.html)
+- [`SELECT`](select.html)
+- [`SET`](set-vars.html)
+- [`SET CLUSTER SETTING`](set-cluster-setting.html)
+- [`SHOW`](sql-grammar.html#show_stmt)
+- [`UPDATE`](update.html)
+- [`UPSERT`](upsert.html)
```
Comment (knz) [inline v2.0/explain.md]: Add EXPLAIN to this list.

File: v1.1/explain.md [hunk 6]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`PAUSE JOB`](sql-grammar.html#pause_stmt)
+- [`RESET`](sql-grammar.html#reset_stmt)
+- [`RESTORE`](restore.html)
+- [`RESUME JOB`](sql-grammar.html#resume_stmt)
+- [`SELECT`](select.html)
+- [`SET SESSION`](sql-grammar.html#set_session_stmt)
+- [`SET CLUSTER SETTING`](set-cluster-settings.html)
+- [`SHOW`](sql-grammar.html#show_stmt)
+- [`UPDATE`](update.html)
+- [`UPSERT`](upsert.html)
```
Comment (knz) [inline v1.1/explain.md]: Add EXPLAIN to this list.

File: v1.0/explain.md [hunk 7]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable statements
+
+You can use `EXPLAIN` on the following statements:
+
+- [`ALTER TABLE`](alter-table.html)
+- [`CREATE`](sql-grammar.html#create_stmt)
```
Comment (knz) [inline v1.0/explain.md]: link to the doc pages for create instead of the grammar.

File: v1.1/explain.md [hunk 8]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable Statements
+
+- [`ALTER`](sql-grammar.html#alter_stmt)
+- [`BACKUP`](sql-grammar.html#backup_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link backup.html

File: v1.0/explain.md (hunk 3 above)
Comment (knz) [reply v1.0/explain.md]: I would also put the various ALTER statements on the same line with proper doc links like this:

```
- [`ALTER TABLE`](alter-table.html), [`ALTER INDEX`](alter-index.html), ...
```

File: v1.1/explain.md [hunk 9]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable Statements
+
+- [`ALTER`](sql-grammar.html#alter_stmt)
```
Comment (knz) [inline v1.1/explain.md]:
```
- [`ALTER TABLE`](alter-table.html), [`ALTER INDEX`](alter-index.html), ...
```

File: v1.1/explain.md [hunk 10]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`BACKUP`](sql-grammar.html#backup_stmt)
+- [`CANCEL`](sql-grammar.html#cancel_stmt)
```
Comment (knz) [inline v1.1/explain.md]:
```
- [`CANCEL JOB`](cancel-job.html), [`CANCEL QUERY`](cancel-query.html)
```

File: v1.1/explain.md [hunk 11]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`CREATE`](sql-grammar.html#create_stmt)
```
Comment (knz) [inline v1.1/explain.md]: ditto various creates

File: v1.1/explain.md [hunk 12]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`DELETE`](delete.html)
+- [`DROP`](sql-grammar.html#drop_stmt)
```
Comment (knz) [inline v1.1/explain.md]: ditto various drops

File: v1.1/explain.md [hunk 13]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`EXECUTE`](sql-grammar.html#execute_stmt)
+- [`IMPORT`](sql-grammar.html#import_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link import.html

File: v1.1/explain.md [hunk 14]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`PAUSE JOB`](sql-grammar.html#pause_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link pause-job.html

File: v1.1/explain.md [hunk 15]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`RESET`](sql-grammar.html#reset_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link reset-vars.html

File: v1.1/explain.md [hunk 16]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`RESTORE`](restore.html)
+- [`RESUME JOB`](sql-grammar.html#resume_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link resume-job.html

File: v1.1/explain.md [hunk 17]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`SELECT`](select.html)
+- [`SET SESSION`](sql-grammar.html#set_session_stmt)
```
Comment (knz) [inline v1.1/explain.md]: link set-vars.html

File: v1.1/explain.md [hunk 18]
```diff
@@ -4,10 +4,31 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
+- [`SET CLUSTER SETTING`](set-cluster-settings.html)
+- [`SHOW`](sql-grammar.html#show_stmt)
```
Comment (knz) [inline v1.1/explain.md]: list the various SHOW pages

File: v2.0/explain.md [hunk 19]
```diff
@@ -4,10 +4,32 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
...
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable Statements
+
+- [`ALTER`](sql-grammar.html#alter_stmt)
```
Comment (knz) [inline v2.0/explain.md]: ditto precise page links as for 1.1.

File: v2.0/explain.md [hunk 20]
```diff
@@ -4,10 +4,32 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
 toc: false
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
 
 <div id="toc"></div>
 
+## Explainable Statements
+
```
Comment (jseldess) [inline v2.0/explain.md]: Let's us an intro sentence like you have in the 1.1 version of the page:

> You can use `EXPLAIN` on the following statements:

File: v1.0/explain.md [hunk 21]
```diff
@@ -4,10 +4,24 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
 toc: false
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute the [explainable statements](#explainable-statments). You can then use this information to optimize those queries.
```
Comment (jseldess) [inline v1.0/explain.md]: nit: `to execute the explainable statements` > `for an explainable statement`.

Also, I'd tweak the second sentence a bit: `You can then use this information to optimize the query.`

Comment (jseldess) [reply v1.0/explain.md]: Please apply this change to other versions of the page as well.

File: v1.0/explain.md (hunk 1 above)
Comment (lhirata) [reply v1.0/explain.md]: Done

File: v1.0/explain.md (hunk 2 above)
Comment (lhirata) [reply v1.0/explain.md]: Done

File: v1.0/explain.md (hunk 3 above)
Comment (lhirata) [reply v1.0/explain.md]: According to the full SQL grammar, only ALTER TABLE is explainable in 1.0. I'll update the following versions with how you suggested though.

File: v1.0/explain.md (hunk 4 above)
Comment (lhirata) [reply v1.0/explain.md]: Done

File: v2.0/explain.md (hunk 5 above)
Comment (lhirata) [reply v2.0/explain.md]: Done

File: v1.1/explain.md (hunk 6 above)
Comment (lhirata) [reply v1.1/explain.md]: Done

File: v1.1/explain.md (hunk 8 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 9 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 10 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 13 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 14 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 15 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 16 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 17 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v1.1/explain.md (hunk 18 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v2.0/explain.md (hunk 19 above)
Comment (lhirata) [reply v2.0/explain.md]: done

File: v2.0/explain.md (hunk 20 above)
Comment (lhirata) [reply v2.0/explain.md]: done

File: v1.0/explain.md (hunk 21 above)
Comment (lhirata) [reply v1.0/explain.md]: done

File: v1.0/explain.md [hunk 22]
```diff
@@ -4,10 +4,23 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
 toc: false
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan for an [explainable statements](#explainable-statements). You can then use this information to optimize the query.
```
Comment (jseldess) [inline v1.0/explain.md]: `explainable statements` > `explainable statement`

File: v1.1/explain.md [hunk 23]
```diff
@@ -4,10 +4,34 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
 toc: false
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan for an [explainable statements](#explainable-statements). You can then use this information to optimize the query.
```
Comment (jseldess) [inline v1.1/explain.md]: `explainable statements` > `explainable statement`

File: v2.0/explain.md [hunk 24]
```diff
@@ -4,10 +4,35 @@ summary: The EXPLAIN statement provides information you can use to optimize SQL
 toc: false
 ---
 
-The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan to execute [`DELETE`](delete.html), [`INSERT`](insert.html), [`SELECT`](select.html) or [`UPDATE`](update.html) statements. You can then use this information to optimize those queries.
+The `EXPLAIN` [statement](sql-statements.html) returns CockroachDB's query plan for an [explainable statements](#explainable-statements). You can then use this information to optimize the query.
```
Comment (jseldess) [inline v2.0/explain.md]: `explainable statements` > `explainable statement`

File: v1.1/explain.md (hunk 23 above)
Comment (lhirata) [reply v1.1/explain.md]: done

File: v2.0/explain.md (hunk 24 above)
Comment (lhirata) [reply v2.0/explain.md]: done

**Comment by jseldess** [comment]
LGTM, @lhirata!

**Pull request merged**